### PR TITLE
Allow editing of telegram

### PR DESCRIPTION
**Pull request by SAN-MUYUN**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -44,14 +45,16 @@ public class EditCommand extends Command {
...
+            + "[" + PREFIX_MODULE + "MODULE]... "
+            + "[" + PREFIX_TELEGRAM + "TELEGRAM] \n"
             + "Example: " + COMMAND_WORD + " 1 "
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com";
 
     public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+    public static final String MESSAGE_DUPLICATE_TELEGRAM = "This telegram already exists in the address book.";
```
Comment (darrenchooji) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:57]: Small nitpicking from me, but perhaps can say, "This telegram handle already ..."

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -56,10 +57,10 @@ public void execute_someFieldsSpecifiedUnfilteredList_success() {
 
         PersonBuilder personInList = new PersonBuilder(lastPerson);
         Person editedPerson = personInList.withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
-                .withTags(VALID_TAG_HUSBAND).build();
+                .withTags(VALID_TAG_HUSBAND).withTelegram(VALID_TELEGRAM_BOB).build();
```
Comment (darrenchooji) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:60]: Might want to consider removing ".withTags". I am pretty sure it is no longer used once merged with Wallace's PR.

**Review by darrenchooji — COMMENTED** [review]
Overall, LGTM. Some minor nitpicks with the test case and phrasing of message.

**Review by LonelyFort — COMMENTED** [review]
Might need to redesign getter and setter inside EditCommand to not introduce nullpointerexception.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -221,6 +231,30 @@ public Optional<Set<Module>> getModules() {
...
+        /**
+         * Returns a new Optional object that contains the {@code telegram}.
+         * Returns {@code Optional#empty()} if {@code telegram} is empty.
+         */
+        public Optional<Telegram> getTelegram() {
+            return (telegram != null && telegram.isPresent())
+                    ? Optional.of(telegram.get())
+                    : telegram == null
+                    ? null
+                    : Optional.empty();
```
[the remark this answers is not in the document]
Comment (SAN-MUYUN) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:255]: No it will not. It is meant to be like that

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (TeaMakesMePee) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:57]: I have already added this in my PR.

**Review by TeaMakesMePee — COMMENTED** [review]
Everything else looks good, just ensure enough test coverage

**Review by darrenchooji — COMMENTED** [review]
Overall everything still LGTM! Might want to consider adding more test cases for duplicate handling of Telegram handles or name from editing.
